src/core/store.ts

```diff
@@ -26,20 +26,65 @@ type Revision = number
 type InvalidatedRevision = number
 type ReadDependencies = Map<AnyAtom, Revision>
 
-// immutable atom state
+/**
+ * Immutable atom state, tracked for both mounted and unmounted atoms in a store.
+ */
 export type AtomState<Value = unknown> = {
+  /**
+   * Counts number of times atom has actually changed or recomputed.
+   */
   r: Revision
+  /**
+   * Marks the revision of this atom when a transitive dependency was invalidated.
+   * Mounted atoms are considered invalid when `r === i`. Unmounted atoms are
+   * always considered invalid.
+   */
   i?: InvalidatedRevision
+  /**
+   * Maps from a dependency to the dependency's revision when it was last read.
+   * We can skip recomputation of an atom by comparing the ReadDependencies revision
+   * of each dependency to that dependencies's current revision.
+   */
   d: ReadDependencies
 } & ({ e: ReadError } | { p: SuspensePromise } | { v: ResolveType<Value> })
 
-export type VersionObject = { p?: VersionObject } // "p"arent version
+/**
+ * Represents a version of a store. A version contains a state for all atoms.
```

Comment:
To be precise, it's not for *all* atoms. We only copy atom states on demand, so it will contain a subset of all atoms.